src/Core/src/Hosting/Dispatching/AppHostBuilderExtensions.cs

```diff
@@ -29,6 +34,98 @@ public static MauiAppBuilder ConfigureDispatching(this MauiAppBuilder builder)
 			return builder;
 		}
 
+		public static MauiAppBuilder ConfigureEnvironmentVariables(this MauiAppBuilder builder)
+		{
+			IDictionary environmentVariables = Environment.GetEnvironmentVariables();
+
+#if !NETSTANDARD
+			// For Android we read the environment variables from a text file that is written to the device/emulator
+			if (OperatingSystem.IsAndroid())
+			{
+				var envVarLines = System.IO.File.ReadAllLines("/data/local/tmp/ide-launchenv.txt");
+
+				environmentVariables = envVarLines
```

Comment:
Do we want to overwrite the variables captured above?  My first thought would be we want to combine (and prefer any values from this file over any existing), not completely replace them with these?

Reply:
@emaf has done the work in the tooling for this, forwarding the environment variables from the host development machine to the emulator/Simulator/actual device. Right now this happens through the Hot Reload connect but for Android that is not always very reliable. So for Android what is also done is write the environment variables to this file and we can pick them up from there directly. There should not be any information we're missing by overwriting this.

While typing this, I do realize that we probably have the requirement of having Hot Reload enabled for this functionality. We might want to look into if we can make that better.